umbra-js/src/utils/utils.ts

```diff
@@ -378,22 +384,51 @@ export async function getEthSweepGasInfo(
     : await provider.estimateGas({ gasPrice: 0, to, from, value: fromBalance });
 
   // Estimate the gas price, defaulting to the given one unless on a network where we want to use provider gas price
-  const gasPrice = ignoreGasPriceOverride.includes(chainId)
+  let gasPrice = ignoreGasPriceOverride.includes(chainId)
     ? providerGasPrice
     : BigNumber.from((await overrides.gasPrice) || providerGasPrice);
 
+  // On networks with EIP-1559 gas pricing, the provider will throw an error and refuse to submit
+  // the tx if the gas price is less than the block-specified base gas cost. The error is "max fee
+  // per gas less than block base fee". So we need to ensure that the low estimate we're using isn't
+  // *too* low.  Additionally, if the previous block exceeded the target size, the base fee will be
+  // increased by 12.5% for the next block, per:
+  //   https://ethereum.org/en/developers/docs/gas/#base-fee
+  // To be conservative, therefore, we assume a 12.5% increase will affect the base fee for the
+  // transaction we're about to send.
+  const baseGasFee = (lastBlockData?.baseFeePerGas || BigNumber.from('1')).mul('1125').div('1000');
+  if (gasPrice.lt(baseGasFee)) gasPrice = baseGasFee;
```

Comment:
RR. I'm a bit confused here, isn't this saying that if the block had no base fee (i.e. EIP-1559 isn't supported) then use a base fee of 1.125 gwei? I think what we want is:

```suggestion
// Get basefee from block, or zero if EIP-1559 is not supported.
const baseFeePerGas = lastBlockData?.baseFeePerGas || BigNumber.from('0')
// If the gas price is smaller than the base fee, increase it to match the base fee
if (gasPrice.lt(baseGasFee)) gasPrice = baseGasFee;
// Add 12.5% to account for gas price increase
gasPrice = gasPrice.mul(1125).div(1000);
```

Reply:
Ah, you're right. That should have been zero.

But I don't think we want to apply the 12.5% increase to the gasPrice if we're not using the base fee for the gasPrice. Really what I think we want is: if gasPrice < baseGasFee, then gasPrice = baseGasFee * 1.125. I'll make this change

Reply:
Fixed with 948cdd8adb84cf4453ac5409c70ca93d70195693

Reply:
> The 12.5% increase is just applied to the base fee because we pessimistically assume that it will be increased in the next block. I

Yea, but `gasPrice` can also increase in the next block, so wouldn't we want to add margin there also? At the very least, it ensures the tx gets mined faster which is nicer UX

>  If the gasPrice that the provider predicts is above the pessimistic base fee, it should be safe to use without padding it also

What about networks without base fees? Gas price can increase in the next block still

Reply:
> gasPrice can also increase in the next block, so wouldn't we want to add margin there also?

Good point. And actually I'd forgotten that we are [already padding gas prices](https://github.com/ScopeLift/umbra-protocol/pull/390/files?diff=unified&w=0#diff-83ca539312493c7e5d6efd2715eef9cf3c42f8fea27cd423ea611c4a48ebbe5eR402-R412) globally by 5% or more depending on the chain right below this code. So for that reason I don't think we need to do more

Reply:
Tracking here: https://github.com/ScopeLift/umbra-protocol/issues/477